packages/lib/modules/pool/PoolList/PoolListTokenPills.tsx

```diff
@@ -190,10 +190,10 @@ export function PoolListTokenPills({
     token => token.address !== pool.address
   ) as GqlPoolTokenDetail[]
 
-  if (isV3Pool(pool) && pool.hasErc4626 && !pool.hasNestedErc4626) {
-    // TODO: Move this into a general 'displayTokens' helper function.
+  // TODO: Move this into a general 'displayTokens' helper function.
+  if (isV3Pool(pool) && pool.hasErc4626 && pool.hasAnyAllowedBuffer) {
     poolTokens = poolTokens.map(token =>
-      token.underlyingToken
+      !token?.underlyingToken?.isBufferAllowed
         ? ({ ...token, ...token.underlyingToken } as unknown as GqlPoolTokenDetail)
         : token
```

Comment:
```suggestion
      token?.underlyingToken?.isBufferAllowed
        ? ({ ...token, ...token.underlyingToken } as unknown as GqlPoolTokenDetail)
        : token
```

Reply:
There were some API changes. Fixed to use `token.isBufferAllowed` instead of `underlyingToken.isBufferAllowed`